Arrowgene.Ddon.GameServer/Handler/QuestDeliverItemHandler.cs

```diff
@@ -31,31 +31,35 @@ public override S2CQuestDeliverItemRes Handle(GameClient client, C2SQuestDeliver
 
             Dictionary<uint, CDataDeliveredItem> deliveredItems = new Dictionary<uint, CDataDeliveredItem>();
             List<CDataItemUpdateResult> itemUpdateResults = new List<CDataItemUpdateResult>();
-            foreach (var item in request.ItemUIDList)
+            Server.Database.ExecuteInTransaction(connection =>
             {
-                uint itemId = Server.ItemManager.LookupItemByUID(Server, item.UId);
-                var searchResult = client.Character.Storage.FindItemByUIdInStorage(ItemManager.BothStorageTypes, item.UId);
-                var itemUpdate = Server.ItemManager.ConsumeItemByUId(Server, client.Character, searchResult.Item1, item.UId, item.Num) 
-                    ?? throw new ResponseErrorException(ErrorCode.ERROR_CODE_QUEST_DONT_HAVE_DELIVERY_ITEM);
+                foreach (var item in request.ItemUIDList)
+                {
+                    uint itemId = Server.ItemManager.LookupItemByUID(Server, item.UId);
+                    var searchResult = client.Character.Storage.FindItemByUIdInStorage(ItemManager.BothStorageTypes, item.UId);
+                    var itemUpdate = Server.ItemManager.ConsumeItemByUId(Server, client.Character, searchResult.Item1, item.UId, item.Num, connectionIn: connection)
+                        ?? throw new ResponseErrorException(ErrorCode.ERROR_CODE_QUEST_DONT_HAVE_DELIVERY_ITEM);
```

Comment:
Is this error actually happening? Seems like an issue if it is.

Reply:
This seems like a problem we should try to solve sooner rather than later. Wonder what is causing it.